generate-readme-tables.js

```diff
@@ -0,0 +1,488 @@
+    return accumulation;
+  }
+  if (accumulation.length == 0 && next instanceof Comment) {
+    accumulation.push(next);
+    return accumulation;
+  }
+  if (lastElement instanceof Comment && next instanceof PropertyDeclaration) {
+    accumulation.pop();
+    accumulation.push(new TableEntry(next, lastElement));
+    return accumulation;
+  }
+  if (lastElement instanceof TableEntry && next instanceof PropertyDeclaration) {
+    accumulation.push(new TableEntry(next, new Comment(Constants.UNKNOWN_DESCRIPTION)));
+    return accumulation;
+  }
+
+  if (lastElement instanceof TableEntry && next instanceof Comment) {
+    accumulation.push(next);
+    return accumulation;
+  }
+
+  // should never happen
+  if (lastElement instanceof PropertyDeclaration && next instanceof PropertyDeclaration) {
+    accumulation.push(new TableEntry(next, new Comment(Constants.UNKNOWN_DESCRIPTION)));
+    return accumulation;
+  }
+
+  if (lastElement instanceof PropertyDeclaration && next instanceof Comment) {
+    // push the comment so that it gets picked up next recurse
+    accumulation.push(next);
+    return accumulation;
+  }
+
+  return accumulation;
+}
+
+function constructTableEntries(tokens) {
+  return tokens.map((block) => {
+    let innerTokens = block[1];
+    let tableEntries = innerTokens.reduce(tableEntryReducer, []);
+    return [block[0], tableEntries];
+  });
+}
+
+class PropertyDeclaration {
+  constructor(name, type, description) {
+    this.name = name;
+    this.type = type;
+  }
+}
+
+class Comment {
+  constructor(text) {
+    this.text = text
+  }
+}
+
+class TableEntry {
+  constructor(declaration, comment) {
+    this.declaration = declaration;
+    this.comment = comment;
+  }
+}
+
+// Lets us specify when we're done with styling a piece of text with ANSI escape codes
+const Styling = {
+    // Important - we need to end each line with `FINISHED` otherwise colors and styles will spill.
+    FINISHED: '\033[0m',
+}
+
+// Provides terminal styling of text backgrounds via ANSI escape codes
+const Background = {
+    BLACK: '\u001b[40m',
+    RED: '\u001b[41m',
+    GREEN: '\u001b[42m',
+    YELLOW: '\u001b[43m',
+    BLUE: '\u001b[44m',
+    MAGENTA: '\u001b[45m',
+    CYAN: '\u001b[46m',
+    WHITE: '\u001b[47m',
+}
+
+const Foreground = {
+    // Standard Colors
+    BLACK: '\u001b[30m',
+    WHITE: '\u001b[37m',
+    BLUE: '\033[94m',
+    CYAN: '\033[96m',
+    GREEN: '\033[92m',
+    YELLOW: '\033[93m',
+    RED: '\033[91m',
+    MAGENTA: '\u001b[35m',
+
+    // Bright Colors
+    BRIGHT_WHITE: '\u001b[37;1m',
+    BRIGHT_BLACK: '\u001b[30;1m',
+    BRIGHT_RED: '\u001b[31;1m',
+    BRIGHT_GREEN: '\u001b[32;1m',
+    BRIGHT_YELLOW: '\u001b[33;1m',
+    BRIGHT_BLUE: '\u001b[34;1m',
+    BRIGHT_MAGENTA: '\u001b[35;1m',
+    BRIGHT_CYAN: '\u001b[36;1m',
+
+    // Style Attributes
+    BOLD: '\033[1m',
+    UNDERLINE: '\033[4m',
+}
+
+const Constants = {
+  UNKNOWN_DESCRIPTION: "???",
+  TABLE_HEADER_NAME_TITLE: "Name",
```

Comment:
These've gotta change to match the existing readme.
Also, probably good to hoist them to the top of the file lol